lib/atat-pipeline-stack.ts

```diff
@@ -0,0 +1,47 @@
+import * as cdk from "aws-cdk-lib";
+import * as pipelines from "aws-cdk-lib/pipelines";
+import * as ssm from "aws-cdk-lib/aws-ssm";
+import { Construct } from "constructs";
+import { AtatWebApiStack } from "./atat-web-api-stack";
+import { AtatIamStack } from "./atat-iam-stack";
+import { AtatNetStack } from "./atat-net-stack";
+
+export interface AtatProps {
+  environmentName: string;
+  vpcCidr?: string;
+}
+
+export interface AtatPipelineStackProps extends cdk.StackProps, AtatProps {
+  branch: string;
+  repository: string;
+  githubPatName: string;
+}
+
+export class AtatApplication extends cdk.Stage {
+  constructor(scope: Construct, id: string, props: cdk.StageProps & AtatProps) {
+    super(scope, id, props);
+    const iam = new AtatIamStack(this, `${props.environmentName}Iam`);
+    const net = new AtatNetStack(this, `${props.environmentName}Net`, { vpcCidr: props.vpcCidr });
+    const atat = new AtatWebApiStack(this, "Atat", {
+      environmentName: props.environmentName,
+      network: net,
+    });
+  }
+}
+
+export class AtatPipelineStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props: AtatPipelineStackProps) {
+    super(scope, id, props);
+
+    const pipeline = new pipelines.CodePipeline(this, "Pipeline", {
```

Comment:
Since we are using s3, should we include:
`selfMutation: true` ?
v2 docs: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines-readme.html#s3-error-access-denied

ShellStep docs explicitly setting `selfMutation` to `true` or `false`: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines.ShellStep.html#example

Reply:
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines.CodePipeline.html#selfmutation

> selfMutation?
> Type: boolean (optional, default: true)

We should be good to go here.